### ISSUE TITLE
No documentation is displayed in Hackage

### ISSUE THREAD

**Issue by hackeryarn** [agrafix/users#16]
I think that generating the docs for Hackage would make navigation much easier compared to navigating the source code. 

Current state for users:
![users](https://cloud.githubusercontent.com/assets/827709/13095043/8add6fb0-d4d4-11e5-9549-616a67ef7874.png)

Current state for users-persistent:
![users-persistent](https://cloud.githubusercontent.com/assets/827709/13095018/5cabeb9e-d4d4-11e5-9378-0e34b8149637.png)

**Comment by agrafix** [comment]
Yes, ususally hackage generates this automatically? I will check and upload them otherwise...
